backend/src/facebook-feed/facebook-feed.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { ConfigService } from '../config/config.service';
+import sources from './facebook-feed-sources';
+import { InjectRepository } from '@nestjs/typeorm';
+import { TransactionRepository, Repository, Transaction } from 'typeorm';
+import { FacebookFeed } from './facebook-feed.entity';
+const config = new ConfigService();
+const axios = require('axios');
+
+@Injectable()
+export class FacebookFeedService {
+
+    constructor(
+        @InjectRepository(FacebookFeed)
+        @TransactionRepository(FacebookFeed)
+        private readonly facebookFeedRepository: Repository<FacebookFeed>,
+        private readonly logger: Logger) {
+        this.logger = new Logger('FacebookFeedService');
+    }
+
+    async onModuleInit() {
+        await this.fetchFacebookFeeds();
+    }
+
+    @Transaction()
+    async fetchFacebookFeeds() {
+        try {
+            const vantaaFacebookDataUrl = `https://graph.facebook.com/v3.1/${sources.vantaa}/posts?`
+                + `fields=id,name,message,story,picture,message_tags,story_tags,caption,`
+                + `created_time,description,place,properties,status_type,type,comments,reactions`
+                + `&access_token=${config.facebookPageToken}`;
+            const facebookFeeds = this.transformData((await axios.get(vantaaFacebookDataUrl)).data.data);
```

Comment:
probably have a retrieval function here instead of spreading everything in place.

```javascript
const queryObj = {...} // the object query you have for fb request, forexample fields, created_time, etc..
const configuredUrlRequest = this.configService.FB_FEED_BASE_URL + utils.parseToQueryParams(queryObj)

fetchData(configuredUrlRequest).then(transformFunction).then(presistToDb)
```

You can apply this in a easier to read step by step algorithm.
1) Specify params to send
2) Configure url
3) Fetch data and apply transform
4) Send it off to somewhere else (other codes, database, consumers)

This has the benefit of
1) Easy to change query params, and the parsing function can be tested, so you dont make silly mistakes in typos
2) Specification of query params is not muddled with the query itself. => Increased readability
3) Data is fetch, and transformed in a easy to read manner. Fetch, then transform, then use.

I suggest Promise's then native function instead of await, await, await because it improves readability, and is easier to read as a chain of actions instead of intermediates statements.

A general rule of thumbs
1) When you need to coordinate, use await
2) When you have step by step transformation or usage of a piece of data, use `then`